src/darwin/Framework/CHIP/MTROperationalCertificateIssuer.h

```diff
@@ -0,0 +1,91 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Matter/MTRAttestationInfo.h>
+#import <Matter/MTRCSRInfo.h>
+#import <Matter/MTRCertificates.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class MTRDeviceController;
+
+/**
+ * A representation of the operation certificate info for a node.
+ *
+ * A nil intermediateCertificate means there is no intermediate.
+ *
+ * adminSubject is passed to the device as part of the AddNOC command.  A nil
+ * adminSubject means the node id of the relevant MTRDeviceController will be
+ * used.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTROperationalCertificateInfo : NSObject
+@property (nonatomic, copy) MTRCertificateDERBytes operationalCertificate;
+@property (nonatomic, copy, nullable) MTRCertificateDERBytes intermediateCertificate;
+@property (nonatomic, copy) MTRCertificateDERBytes rootCertificate;
+@property (nonatomic, copy, nullable) NSNumber * adminSubject;
+
++ (instancetype)infoWithOperationalCertificate:(MTRCertificateDERBytes)operationalCertificate
```

Comment:
Is there some particular reason for using a factory method instead of an initializer here?

Reply:
Not really; I wasn't sure which was the more idiomatic thing to do, or both.

Reply:
I think initializer would be the default. I'd generally use a factory method if you might want to return a cached instance, or return an instance of a particular sub-class depending on parameters, that sort of thing.

Reply:
Done.